sample/transforms.go

```diff
@@ -0,0 +1,137 @@
+package sample
+
+import (
+	"cmp"
+	"errors"
+	"math"
+	"slices"
+
+	pq "github.com/emirpasic/gods/v2/queues/priorityqueue"
+	"gonum.org/v1/gonum/floats"
+)
+
+type Transform interface {
+	Apply([]float64) ([]float64, error)
+}
+
+// TODO(parthsareen): potentially cache softmax values
+func softmax(logits []float64) []float64 {
+	var sum float64
+	probs := make([]float64, len(logits))
+	for i, v := range logits {
+		probs[i] = math.Exp(v)
+		sum += probs[i]
+	}
+	floats.Scale(1/sum, probs)
```

Comment:
I don't believe we need the `floats` dependency. Could we just do something like this?

```
for i := range probs {
    probs[i] /= sum
}
```